spring-rabbit/src/test/java/org/springframework/amqp/rabbit/connection/RoutingConnectionFactoryTests.java

```diff
@@ -323,4 +324,23 @@ protected synchronized void redeclareElementsIfNecessary() {
 		assertThat(SimpleResourceHolder.unbind(connectionFactory)).isEqualTo("foo");
 	}
 
+	@Test
+	void afterPropertiesSetShouldNotThrowAnyExceptionAfterAddTargetConnectionFactory() throws Exception {
+		AbstractRoutingConnectionFactory routingFactory = new AbstractRoutingConnectionFactory() {
+			@Override
+			protected Object determineCurrentLookupKey() {
+				return null;
+			}
+		};
+
+		routingFactory.addTargetConnectionFactory("1", Mockito.mock(ConnectionFactory.class));
+
+		try {
+			routingFactory.afterPropertiesSet();
+		}
+		catch (Exception e) {
+			fail("afterPropertiesSet should not throw any exception after addTargetConnectionFactory", e);
```

Comment:
We prefer to use `Assertions.assertThatNoException()` for this type of assertions instead.